### ISSUE TITLE
Design and accessiblity review: Create and view DNS records (I Beam)

### ISSUE THREAD

**Issue by erinysong** [cisagov/manage.get.gov#4631]
### Sub-issue description

At this point in our DNS prototype, we can view and create DNS records. 

The related features and functionality have already been reviewed by engineers and now we need designers to review them, as well.

There are accessibility issues found during dev review that are also documented in this ticket. They should be addressed in this ticket as well.

### Acceptance criteria

- [ ] Design and accessibility review the DNS record management on the DNS records page
- [ ]  Document any changes needed in the Slack channel using our established norms (see Slack channel instructions)
- [ ]  Discuss changes with dev and determine which are small enough to be done as part of this ticket
- [ ] Work with dev to create new tickets for larger changes

### Additional context

Working with dev: Dev effort on this should be scoped to a "2" ticket

Dev is responsible for answering questions from design, making minor changes (such as those that we would normally make on a PR).
Anything that would take more than ~4 hours to complete alone should be a new issue.
If the changes desired will result in the dev spending more than a full day on all changes, make a dev ticket for items where reasonable (communicate with dev leads).

**Temporary Slack channel (tbd)**
Discussions for this design review should take place in this Slack channel: <insert link when Slack channel created>
